### DISCUSSION TITLE
How to check if SSE socket is closed?

### DISCUSSION THREAD

**Discussion by telenkov88** [miguelgrinberg/microdot#231] [Q&A]
```
@app.route('/events')
@with_sse
async def events(request, sse):
    while True:
        await asyncio.sleep(1)
        await sse.send({'keepalive': True})
        
```

This code never finished, resulting leaks and unreachable server.

I need something like:

```
@app.route('/events')
@with_sse
async def events(request, sse):
    while True:
        if sse.isalive:
            await asyncio.sleep(1)
            await sse.send({'keepalive': True})
        else:
            print("connection closed")
            break
```

**Comment by miguelgrinberg** [comment]
Hmm. This may be an oversight on my part. The `sse.send()` call should raise an exception when the other side is gone. I'll check and make sure this mechanism is in place.

**Reply by telenkov88** [reply]
Thank you, I just bumped to this problem on esp32 where number of connections is limited by 10.
